#25 Singly linked lists vs doubly linked lists

andrew-zhouu opened the pull request:
#25

andrew-zhouu committed:
#24 Singly linked lists and doubly linked lists

initial commit

andrew-zhouu committed:
#24 Singly linked lists and doubly linked lists

Change titles

andrew-zhouu committed:
#24 Singly linked lists and doubly linked lists

Added Works Cited

andrew-zhouu committed:
#24-Singly-linked-lists-and.md

Changed to markdown document

andrew-zhouu committed:
#24-Singly-linked-lists-and.md

Cleaned up Works Cited

langrh commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,39 @@
+Similar to the doubly linked list, singly linked lists can also be used to undo functions. Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. 
+Singly linked lists can also be used to represent polynomials and graphs because of the links between different values in polynomials.
+
+### Conclusion
```
Elaborating more on the functions that can be created would be good. Maybe just a couple examples.

langrh reviewed (commented):
This goes in great detail explaining singly linked lists and doubly linked lists and comparing them, with great real work examples. I feel like I understand what singly linked lists and doubly linked lists are now and the differences between them. Over all, very interesting.

justaskdavidb committed:
Update #24-Singly-linked-lists-and.md

Moving this ticket into ready to review and approving it. @KellynBE, please review and check for where I have made grammatical errors.

justaskdavidb reviewed (dismissed):
LGTM

justaskdavidb commented:
@langrh, thanks for reviewing this PR!

dkbaldwin committed:
Merge branch 'main' into 24-Singly-Linked-Lists-Vs-Doubly-Linked-Lists

justaskdavidb committed:
Merge branch 'main' into 24-Singly-Linked-Lists-Vs-Doubly-Linked-Lists

justaskdavidb commented:
@KellynBE please review this PR

KellynBE commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,24 @@
+Singly linked lists and doubly linked lists are special types of data structure because they introduce the concept of pointers (links) and nodes. A linked list is a collection of similar items called nodes that seek to establish a sequence / order to the collection. There are two types of node -- the head and tail nodes and the head is always the first node.  Singly linked lists’ nodes can only a ...
+
+### Singly Linked Lists
+Singly linked lists’ nodes are connected only to the next node. Singly linked lists are great when you want to access data in the forward direction only. The insertion and deletion of nodes of nodes is easy. Singly linked list require less memory than doubly linked lists. However, the main problem with singly linked lists is that directly accessing the previous node is impossible. You experience  ...
```
- change "lists'" to "list"
- reword "when you want to access data in the forward direction only" to "when you want to only access subsequent data"
- Why is insertion and deletion easy? is it because they require less memory?
- delete second "of nodes"
- make "list" plural
- delete the bank sentence
- reword last sentence, something like: "Imagine you're listening to an album, but the rewind button doesn't work. You can listen to the songs in sequential order, but you can't go back to listen to your favorite one again."

KellynBE commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,24 @@
+Singly linked lists’ nodes are connected only to the next node. Singly linked lists are great when you want to access data in the forward direction only. The insertion and deletion of nodes of nodes is easy. Singly linked list require less memory than doubly linked lists. However, the main problem with singly linked lists is that directly accessing the previous node is impossible. You experience  ...
+
+### Doubly Linked Lists
+In contrast, doubly linked lists’ nodes are connected to both the previous and next node and this difference is very important. The main advantage of a doubly linked list over a singly linked list is its ability to traverse easily both in a forward and backward directions. This means you now have the ability ot rewind the song to the beginnging. But doubly linked lists can be disadvantageous comp ...
```
- change "lists'" to "list"
- change "next" to "following"
- delete "and this difference is very important"
- make "directions" singular
- correct spelling of "to" and "beginning"
- reword "But doubly linked lists have their disadvantages as well."
- change "Additionally" to "And" and delete the comma

KellynBE commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,24 @@
+In contrast, doubly linked lists’ nodes are connected to both the previous and next node and this difference is very important. The main advantage of a doubly linked list over a singly linked list is its ability to traverse easily both in a forward and backward directions. This means you now have the ability ot rewind the song to the beginnging. But doubly linked lists can be disadvantageous comp ...
+
+## Conclusion
+Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. A bank teller's line is a perfect everyday example of its advantage; and a broken rewind button its disadvantage. The advantaged of doubly linked lists can be demonstrated when having a song that you want to fast forward or rewind -- a feature that singly linked list would not possess. Doubly l ...
```
- replace second sentence with something like "They're great for moving forward, but not so much moving back, like a stereo with a broken rewind button."
- delete d in "advantaged"
- replace third sentence "A doubly linked list, in contrast, is a stereo with functioning rewind and fast forward buttons."
- I would move the last two sentences under the Doubly Linked List section; this isn't information that should be introduced in the conclusion.
- In sentence beginning "Doubly linked lists..." insert "and redo actions" after undo and delete "because it involves a user’s history"
- delete "Similar to the doubly linked list," and capitalize "singly"
- comma after "functions"
- change "would never" to "wouldn't"

KellynBE commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,24 @@
+Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. A bank teller's line is a perfect everyday example of its advantage; and a broken rewind button its disadvantage. The advantaged of doubly linked lists can be demonstrated when having a song that you want to fast forward or rewind -- a feature that singly linked list would not possess. Doubly l ...
+
+
+### Conclusion
```
Why is there a second conclusion?

jasalise replied:
2nd conclusion has been removed

KellynBE commented on #24-Singly-linked-lists-and.md:
```diff
@@ -0,0 +1,24 @@
+
+
+### Conclusion
+ can be problematic or beneficial depending on the circumstance and each have their own unique uses. Linked lists are very important to computer science, and many commonly used programming concepts such as queues and stacks owe their existance to linked lists.
```
- Beginning of paragraph is missing
- comma after "circumstance"
- commas around "such as queues and stacks"
- do you want the queues and stacks piece links within this article?

dkbaldwin committed:
made 'structure' plural

dkbaldwin committed:
* modified file name
* made some changes to content of the file per the requested changes by Editor

andrew-zhouu committed:
24-singly-doubly-linked-lists.md

Edited and updated grammar issues and suggestions from editor KellynBE

dkbaldwin committed:
Merge branch 'main' into 24-Singly-Linked-Lists-Vs-Doubly-Linked-Lists

jasalise reviewed (dismissed):
Made all of the corrections.

Jandunn commented on singly-doubly-linked-lists.md:
```diff
@@ -0,0 +1,20 @@
+In contrast, doubly linked lists’ nodes are connected to both the previous and next node. The main advantage of a doubly linked list over a singly linked list is its ability to traverse easily both in a forward and backward direction. This means you now have the ability to rewind the song to the beginning. Doubly linked lists are often used to undo and redo operations and can be used to track or  ...
+
+## Conclusion
+Singly linked lists are implemented in stacks and queues, which are the foundation of computer science. They're great for moving forward, but not so much moving back, like a stereo with a broken rewind button. A doubly linked list, in contrast, is a stereo with functioning rewind and fast forward buttons. Ultimately, both singly linked lists and doubly linked lists can be problematic or beneficia ...
```
Remove 2nd linked lists
Add period after circumstance

jasalise replied:
Changes made

jasalise committed:
#25 Singly linked lists vs doubly linked lists

Remove 2nd linked lists
Add period after circumstance

jasalise committed:
#25 Singly linked lists vs doubly linked lists

corrected selling of computer

justaskdavidb reviewed (approved):
@jasalise  great job on doing your first code review
@Jandunn , thanks for your continued support!

justaskdavidb commented:
@KellynBE , please review to see if your requested changes have been made. If so, please approve and move to medium. Minor modifications can bee made inside medium.

dkbaldwin committed:
Merge branch 'main' into 24-Singly-Linked-Lists-Vs-Doubly-Linked-Lists

dkbaldwin committed:
Merge branch 'main' into 24-Singly-Linked-Lists-Vs-Doubly-Linked-Lists
